src/web/project/src/pages/Project.tsx

```diff
@@ -0,0 +1,447 @@
+import { useEffect, useState } from 'react';
+import { Link, useLocation, useNavigate, useParams } from 'react-router-dom';
+import SyntaxHighlighter from 'react-syntax-highlighter';
+import * as styles from 'react-syntax-highlighter/dist/esm/styles/hljs';
+import { Button, InputGroup, Modal, Form } from 'react-bootstrap';
+import { useDeleteProject, useGetProject, useUpdateProject } from '../api/Queries';
+import { getProjectResponse, updateProjectRequest } from '../api/interfaces';
+import { useFormik } from 'formik';
+import { AxiosResponse } from 'axios';
+import { UseQueryResult } from 'react-query';
+const Project: React.FC = () => {
+    const navigate = useNavigate();
+    const { state } = useLocation();
+    const params = useParams();
+    const [isUpdateModalShowed, setUpdateModal] = useState(false);
+    const [isDelModalShowed, setDeleteModal] = useState(false);
+    const project: UseQueryResult<AxiosResponse<getProjectResponse>> = useGetProject(
+        params.projectId || ''
+    );
+    const deleteProject = useDeleteProject();
+    const updateProject = useUpdateProject();
+    // const passTransactionData (link: string, data: Transaction) => {
+    //     history.pushState
+    // }
+    const formik = useFormik({
+        enableReinitialize: true,
+        initialValues: {
+            name: '',
+            slug: '',
+            description: '',
+            api_base: '',
+            provider_name: '',
+            ai_model_name: '',
+            tags: '',
+            org_id: ''
+        },
+        onSubmit: async ({
+            name,
+            slug,
+            description,
+            api_base,
+            provider_name,
+            ai_model_name,
+            tags,
+            org_id
+        }) => {
+            const reqValues: updateProjectRequest = {
+                id: params.projectId || '',
+                name,
+                slug,
+                description,
+                ai_providers: [
+                    {
+                        api_base,
+                        provider_name,
+                        ai_model_name
+                    }
+                ],
+                tags: tags.replace(/\s/g, '').split(','),
+                org_id
+            };
+            console.log(reqValues);
+            updateProject.mutateAsync(reqValues).then(() => {
+                setUpdateModal((e) => !e);
+                project.refetch();
+            });
+        }
+    });
+    useEffect(() => {
+        if (project.isSuccess) {
+            const data = project.data.data;
+            if (state !== null) {
+                const transactionData = data.transactions.filter((el) => el.id == state);
+                if (transactionData.length > 0) {
+                    console.log(params.projectId);
```

Comment:
jezeli console.logi nie sa potrzebne to je usun

Reply:
zrobione